Packages/com.unity.inputsystem/Documentation~/Migration.md

```diff
@@ -34,202 +44,108 @@ There are scripting symbols defined which allow you to use conditional compilati
+Action-based input refers to reading pre-configured named axes, buttons, or other controls. ([Read more about Action-based input](./Workflow-Actions.md))
 
-```
-// Query a list of all connected Devices (does not allocate; read-only access).
-InputSystem.devices
-
-// Get notified when a Device is added or removed.
-InputSystem.onDeviceChange +=
-    (device, change) =>
-    {
-        if (change == InputDeviceChange.Added || change == InputDeviceChange.Removed)
-        {
-            Debug.Log($"Device '{device}' was {change}");
-        }
-    }
-
-// Find all gamepads and joysticks.
-var devices = InputSystem.devices;
-for (var i = 0; i < devices.Count; ++i)
-{
-    var device = devices[i];
-    if (device is Joystick || device is Gamepad)
-    {
-        Debug.Log("Found " + device);
-    }
-}
-```
-|Input Manager (Old)|Input System (New)|
-|--|--|
-[`Input.GetKey`](https://docs.unity3d.com/ScriptReference/Input.GetKey.html)|[`ButtonControl.isPressed`](../api/UnityEngine.InputSystem.Controls.ButtonControl.html#UnityEngine_InputSystem_Controls_ButtonControl_isPressed) on the corresponding key:
+- In the old Input Manager, these are defined in the **Axes** list, in the **Input Manager** section of the **Project Settings** window. _(Below, left)_
+- In the new Input System, these are defined in the [Actions Editor](ActionsEditor.md), which can be found in the **Input System Package** section of the **Project Settings** window, or by opening an [Action Asset](ActionAssets.md). _(Below, right)_
 
-```
-// Using KeyControl property directly.
-Keyboard.current.spaceKey.isPressed
-Keyboard.current.aKey.isPressed // etc.
+![](Images/InputManagerVsInputActions.png)</br>_On the left, the old Input Manager Axes Configuration window, in Project settings. On the right, the new Input System's [Actions Editor](ActionsEditor.md)._
 
-// Using Key enum.
-Keyboard.current[Key.Space].isPressed
+__Note:__ In some cases for named axes and buttons, the new Input System requires slightly more code than the old Input Manager, but this results in better performance. This is because in the new Input System, the logic is separated into two parts: the first is to find and store a reference to the action (usually done once, for example in your `Start` method), and the second is to read the action (usually done every frame, for example in your `Update` method). In contrast, the old Input Manager used a string-based API to "find" and "read" the value at the same time, because it was not possible to store a reference to a button or axis. This results in worse performance, because the axis or button is looked up each time the value is read.
 
-// Using key name.
-((KeyControl)Keyboard.current["space"]).isPressed
+To find and store references to actions, which can be axes or buttons use [`FindAction`](../api/UnityEngine.InputSystem.InputActionAsset.html#UnityEngine_InputSystem_InputActionAsset_FindAction_System_String_System_Boolean_). For example:
```

Comment:
Might be worth saying 'via the globally available actions set in the projects settings' (and not that you can also reference an individual asset by name if needed and link out to another page for that part)